### DISCUSSION TITLE
Mismatch between payment_details (docs) and paymentDetails in Twig for PDF Invoices

### DISCUSSION THREAD

**Discussion by mbedded** [kimai/kimai#3234] [General]
**Describe the bug**
The given parameter "payment_details" is not working to fill payment details in a custom twig invoice.
I got a link to this source but in my Case i have to use "template.paymentDetails" to get the desired result.

**To Reproduce**
1. Put custom template in `/var/invoices/my_template.pdf.twig` (as docs say)
2. Create your Invoice Template using UI
3. Create a template using twig. As example see the following snippet:

```html
<table class="footer">
        <tr>
            <td style="width: 33%;">
                {{ model.template.company }}<br>
                {{ model.template.address|trim|nl2br }}
            </td>
            <td style="width: 33%;">
                using "paymentDetails":<br>
                {{ model.template.paymentDetails|nl2br }}
            </td>
            <td style="width: 33%;">
                using "payment_details":<br>
                {{ model.template.payment_details|nl2br }}
            </td>
        </tr>
    </table>
```
4. Press Preview and see the PDF
5. See that payment_details is not working as expected
![image](https://user-images.githubusercontent.com/22235508/161107690-a2aed65c-f778-4e25-8248-df453ca5ff19.png)

**Additional context**
 - Kimai version: kimai/kimai2:apache-1.19.1-prod (Docker)
 - PHP version: 8.0.14
 - Browser: Brave - Version 1.36.122 Chromium: 99.0.4844.88 (Official Build) (64-bit)
 - https://github.com/kevinpapst/kimai2/blob/1.19.1/src/Invoice/Hydrator/InvoiceModelDefaultHydrator.php#L56

Images:

Creating invoice Template
![image](https://user-images.githubusercontent.com/22235508/161107356-015b1b8e-9a27-4294-b674-1320684b229d.png)

Open Preview
![image](https://user-images.githubusercontent.com/22235508/161107559-23e97f32-1681-46e8-80fc-4132d305b3b8.png)

**Comment by kevinpapst** [comment]
Please use the debug template as posted in the other issue.

> You can use a debug template to visualize all available variables:
> https://www.kimai.org/documentation/invoices.html#twig-templates

You are mixing things up. You are reading the docs for:
<img width="780" alt="Bildschirmfoto 2022-03-31 um 20 02 36" src="https://user-images.githubusercontent.com/533162/161120630-8a8c9202-8836-44b1-9259-795373c839d1.png">

and trying to use those in the twig template. 

The twig docs needs some adjustments, that is correct. 
In your case you are trying to use a variable the wrong way. You can either call the variables as the debug template tells you to OR you use the real object methods in the twig template. But those are not documented, because they shouldn't be used (they area internal logic). Accidentally they are called more or less the same in some places.

Please use the linked debug template and create an invoice with it. 
Then use the variables that show up in the debug PDF and everything should be good.
